### PR TITLE
Vision test coverage

### PR DESCRIPTION
**Pull request by reinbach**
Increase vision app test coverage

See https://github.com/unicef/etools-issues/issues/1118

### CODE REVIEW

File: EquiTrack/EquiTrack/factories.py [hunk 1]
```diff
@@ -399,7 +399,7 @@ class BusinessRegionFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = publics_models.BusinessRegion
 
-    name = factory.Sequence(lambda n: 'Business Region {}'.format(n))
+    name = factory.Sequence(lambda n: 'Bus. Region {}'.format(n))
```
Comment (domdinicola) [inline EquiTrack/EquiTrack/factories.py]: just curious why?

File: EquiTrack/vision/adapters/manual.py [hunk 2]
```diff
@@ -23,13 +22,12 @@ class ManualDataLoader(VisionDataLoader):
     /endpoint/object_number else
     """
     def __init__(self, country=None, endpoint=None, object_number=None):
-        if not self.URL:
```
Comment (domdinicola) [inline EquiTrack/vision/adapters/manual.py]: this is needed or has to be moved in the else case (L30)

File: EquiTrack/vision/adapters/partner.py [hunk 3]
```diff
@@ -94,114 +94,123 @@ def bad_record(record):
...
-                    apiobj_field = mapped_key in api_obj
-
-                if field == 'blocked':
-                    apiobj_field = mapped_key in api_obj
-
-                if getattr(local_obj, field) != apiobj_field:
-                    logger.debug('field changed', field)
-                    return True
-            return False
+    def _changed_fields(self, local_obj, api_obj):
```
Comment (domdinicola) [inline EquiTrack/vision/adapters/partner.py]: :+1:

File: EquiTrack/vision/adapters/partner.py [hunk 4]
```diff
@@ -94,114 +94,123 @@ def bad_record(record):
...
+            logger.exception(u'Exception occurred during Partner Sync: {}'.format(exp.message))
+        return processed
 
-                processed += 1
-
-            except Exception as exp:
-                logger.exception(u'Exception occurred during Partner Sync: {}'.format(exp.message))
-            return processed
-
+    def _save_records(self, records):
```
Comment (domdinicola) [inline EquiTrack/vision/adapters/partner.py]: :+1:

File: EquiTrack/EquiTrack/factories.py (hunk 1 above)
Comment (reinbach) [reply EquiTrack/EquiTrack/factories.py]: The `name` field on the `BusinessRegion` model is actually limited to 16 chars, and this default lambda value is greater than that

File: EquiTrack/vision/adapters/manual.py (hunk 2 above)
Comment (reinbach) [reply EquiTrack/vision/adapters/manual.py]: I don't see where/how this URL attribute is being set prior to this point. I'm not seeing any adapters seeing this attribute either. I think I'm missing something?

File: EquiTrack/EquiTrack/factories.py (hunk 1 above)
Comment (domdinicola) [reply EquiTrack/EquiTrack/factories.py]: :+1:

File: EquiTrack/vision/adapters/manual.py (hunk 2 above)
Comment (domdinicola) [reply EquiTrack/vision/adapters/manual.py]: by bad.. I didn't see that that was an assignment!
can we replace that also in the father VisionDataLoader?

Comment (reinbach) [reply EquiTrack/vision/adapters/manual.py]: 👍

File: EquiTrack/vision/vision_data_synchronizer.py [hunk 5]
```diff
@@ -22,8 +22,7 @@ class VisionDataLoader(object):
     URL = ''
 
     def __init__(self, country=None, endpoint=None):
-        if not self.URL:
-            self.URL = settings.VISION_URL
+        self.URL = settings.VISION_URL
```
Comment (domdinicola) [inline EquiTrack/vision/vision_data_synchronizer.py]: actually would be possible to set (line 22)
if we want to exclude this behavior (do we?) we should replace self.URL with settings.VISION_URL everywhere

Comment (reinbach) [reply EquiTrack/vision/vision_data_synchronizer.py]: 👍

File: EquiTrack/vision/vision_data_synchronizer.py [hunk 6]
```diff
@@ -19,12 +19,9 @@
 class VisionDataLoader(object):
     # Caveat - this loader probably doesn't construct a correct URL when the synchronizer's GLOBAL_CALL = True).
     # See https://github.com/unicef/etools/issues/1098
-    URL = ''
+    URL = settings.VISION_URL
```
Comment (domdinicola) [inline EquiTrack/vision/vision_data_synchronizer.py]: :1st_place_medal:  finally we found the best solution!

**Comment by ntrncic** [comment]
@reinbach please resolve conflicts and label ready

**Comment by ntrncic** [comment]
and again please resolve conflicts

**Comment by ntrncic** [comment]
@reinbach please resolve conflicts
